core/src/main/scala/org/http4s/QValue.scala

```diff
@@ -98,10 +101,29 @@ object QValue {
   def unsafeFromString(s: String): QValue =
     fromString(s).fold(throw _, identity)
 
+  private[http4s] val parser: Parser[QValue] = {
+    import cats.parse.Parser.{char => ch, _}
+    import cats.parse.Rfc5234._
+
+    val optWS = (crlf.rep.with1 ~ wsp.rep1).rep
+
+    val qValue = string1(ch('0') ~ (ch('.') ~ digit.rep1).rep)
+      .map(
+        QValue
+          .fromString(_)
+          .valueOr(e => throw e.copy(sanitized = "Invalid q-value"))
+      )
+      .orElse1(
+        ch('1') *> (ch('.') *> ch('0').rep).rep.as(One)
+      )
+
+    (ch(';') *> optWS *> ch('q') *> ch('=') *> qValue).orElse(pure(One))
```

Comment:
I forget: is `q` case insensitive?

Reply:
As per: https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html `All content-coding values are case-insensitive.`

Reply:
Changed to `ignoreCaseChar('q')`